pages/04_multitenancy_and_rls.md

```diff
@@ -0,0 +1,68 @@
+
+The above is all that is necessary if your workbooks have 'embedded datasources' (the datasources haven't been published separately). If you do choose to [publish the datasources to take advantage of Tableau Server's Data Server](https://onlinehelp.tableau.com/current/pro/desktop/en-us/publish_datasources.html), the Document API will still work. You will just have to apply the above workflow first to the datasource(s) and then to the workbooks:
+
+First, build a 'template' datasource, publish it, and build a 'template' workbook that points at the published datasource.
+Then, create a script that:
+1) For each tenant, uses the Document API to change the database connection information in the TDS (datasource file), and save a copy.
+2) Uses the [REST API (or Server Client Library)](./03_server_management_and_restapi.md) to publish each tenant's datasource to Server. The datasource will now have a connection string that you can use in the workbooks...
+3) For each tenant, for each workbook, uses the Document API to change the published datasource connection information, and save a copy.
+4) Uses the [REST API (or Server Client Library)](./03_server_management_and_restapi.md) to publish each all of the workbooks to Server.
+
+### User Filters
+If your tenants 'share' a database server, and there is logic in the database that determines which users can see which rows, you can build your datasources and workbooks using 'User Filters' which is a way to filter based on which user is currently logged in to Tableau Server.
+
+Essentially, when connecting to your data in Tableau Desktop you will JOIN the tables containing the data necessary for analaysis to the entitlements table so that each row has a column ([Owner] in the below example) specifying which Tableau Server user it belongs to. Then you can create a calculation `username() = [Owner]` and filter to `True`.
+
+See also:
+Link | Description
+---- | -----------
+[Setting Up User Filters](https://onlinehelp.tableau.com/current/pro/desktop/en-us/publish_userfilters_create.html#dynamic) | Knowledge base article on setting up user filters using the username() method
+[Securing User Filters](https://onlinehelp.tableau.com/current/pro/desktop/en-us/publish_userfilters_create.html#publish-user-filters) | Knowledge base article on creating your user filters at the data source level so that they are tamper-proof
+[Database Entitlements Strategy](https://tableauandbehold.com/2016/03/07/how-to-set-up-your-database-for-row-level-security-in-tableau/) | Blog post on database structure for row-level security
+[Extract-size strategies for RLS](https://tableauandbehold.com/2016/08/08/defusing-row-level-security-in-tableau-data-extracts-before-they-blow-up-part-1/) | Blog post detailing strategies for managing size of Extracts if your entitlements logic requires many one-to-many JOINs
+
+### In-database Security and Initial SQL
```

Comment:
Not really, unless you're using tickets on top of a 'service' account so every user impersonates a single user. I've added a comment in that page suggesting not doing that.